main.cpp

```diff
@@ -63,18 +63,27 @@ class IntersectionController
   {
     northSouth.setSignal(true); //Initially, North and Southbound "go"
     eastWest.setSignal(false);
+    northSouth.addObserver(this);
```

Comment:
the instance of the east-west lights are ignorant to the state of the model (lightdata)